common/navbar.js

```diff
@@ -388,7 +389,24 @@
                             rid: scope.ridSearchTerm
                         });
 
-                        $window.open(url, '_blank');
+                        // try to fetch the resolver link to see if the path resolves before sending the user
+                        // NOTE: .head returns a 405 error (not allowed) for proper RIDs, maybe ask Karl if this is expected
+                        ConfigUtils.getHTTPService().head(url, {}).then(function (response) {
```

Comment:
I think you should send `deriva-client-context` header with this request for proper logging. And then you don't need the client log that you're doing before this.


```javascript
var logObj = ConfigUtils.getContextHeaderParams();

logObj.action = logService.getActionString(logService.logActions.NAVBAR_RID_SEARCH, "", "");
logObj.rid = rid: scope.ridSearchTerm;

headers[ERMrest.contextHeaderName] = logObj;

onfigUtils.getHTTPService().head(url, {headers: headers})
```